src/unescape.ts

```diff
@@ -2,24 +2,37 @@ import { MinimatchOptions } from './index.js'
 /**
  * Un-escape a string that has been escaped with {@link escape}.
  *
- * If the {@link windowsPathsNoEscape} option is used, then square-brace
+ * If the {@link windowsPathsNoEscape} option is used, then square-bracket
  * escapes are removed, but not backslash escapes.  For example, it will turn
  * the string `'[*]'` into `*`, but it will not turn `'\\*'` into `'*'`,
- * becuase `\` is a path separator in `windowsPathsNoEscape` mode.
+ * because `\` is a path separator in `windowsPathsNoEscape` mode.
  *
- * When `windowsPathsNoEscape` is not set, then both brace escapes and
+ * When `windowsPathsNoEscape` is not set, then both square-bracket escapes and
  * backslash escapes are removed.
  *
  * Slashes (and backslashes in `windowsPathsNoEscape` mode) cannot be escaped
  * or unescaped.
+ *
+ * When `magicalBraces` is not set, escapes of braces (`{` and `}`) will not be
+ * unescaped.
  */
 export const unescape = (
   s: string,
   {
     windowsPathsNoEscape = false,
-  }: Pick<MinimatchOptions, 'windowsPathsNoEscape'> = {}
+    magicalBraces = true,
```

Comment:
Note that this is a **different** default than the `escape` function. I really am not a fan of this, but I either had to maintain the existing behavior for each function with different options, or change the behavior.

I'm not entirely sure I made the right decision here. It seems to me that unescape and escape ought to truly mirror each other if called with the same options, but they didn't before this PR and still don't because unescape will unescape redundantly-escaped characters.... but on the other hand, escape used to not escape braces, but unescape would unescape them, so `unescape(escape(pattern))` with `{ magicalBraces: true }` might match different things.